### ISSUE TITLE
BUG: [in version 0.44.0 and 0.45.0] UserWarning: unrecognized nn.Module: Flatten

### ISSUE THREAD

**Issue by Saltsmart** [shap/shap#3643]
### Issue Description

The following code produces `UserWarning: unrecognized nn.Module: Flatten`:

### Minimal Reproducible Example

```python
import torch
from torch import nn, Tensor
from shap import DeepExplainer


X: Tensor = torch.empty(1, 2, 2)
model = nn.Sequential(nn.Flatten(), nn.Linear(4, 1))
e = DeepExplainer(model, X)
print(e.shap_values(X))
```

### Traceback

```shell
UserWarning: unrecognized nn.Module: Flatten
  warnings.warn(f'unrecognized nn.Module: {module_type}')
[[[[0.]
   [0.]]

  [[0.]
   [0.]]]]
```

### Expected Behavior

[[[[0.]
   [0.]]

  [[0.]
   [0.]]]]

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [ ] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

0.44.0 and 0.45.0

**Comment by CloseChoice** [comment]
I guess flatten was never supported, label this as an enhancement
